common/persistence/nosql/nosql_history_store_test.go

```diff
@@ -527,47 +527,630 @@ func TestDeleteHistoryBranch_unusedBranch(t *testing.T) {
+// |
+// | The following is expected: It preserves the remaining nodes for any dependent
+// | branches.
+// |
+// |	Trees
+// |
+// |	┌───────────────────┐    ┌─ ─ ─ ─ ─ ─ ─ ─ ─ ─┐     ┌─ ─ ─ ─ ─ ─ ─ ─ ─ ─┐
+// |	│  Tree: 123        │
+// |	│  Branch: C        │    │  <branch A is     │     │  <branch B is     │
+// |	│  Ancestors:       │        deleted>                  deleted>
+// |	│    Branch: A      │    │                   │     │                   │
+// |	│    EndNode: 3     │
+// |	└───────────────────┘    └─ ─ ─ ─ ─ ─ ─ ─ ─ ─┘     └─ ─ ─ ─ ─ ─ ─ ─ ─ ─┘
+// |
+// |	Nodes
+// |	                         ┌───────────────────┐
+// |	                         │     Tree: 123     │
+// |	                         │     Branch: A     │
+// |	                         │     Node: 1       │
+// |	                         └─────────┬─────────┘
+// |	                                   │
+// |	                                   ┼
+// |	                         ┌─────────▼─────────┐
+// |	                         │     Tree: 123     │
+// |	                         │     Branch: A     │
+// |	                         │     Node: 2       │
+// |	                         └─────────┬─────────┘
+// |	          ┌────────────────────────┘
+// |	          │
+// |	┌─────────▼─────────┐
+// |	│     Tree: 123     │
+// |	│     Branch: C     │
+// |	│     Node: 3       │
+// |	└─────────┬─────────┘
+// |	          │
+// |	          │
+// |	┌─────────▼─────────┐
+// |	│     Tree: 123     │
+// |	│     Branch: C     │
+// |	│     Node: 4       │
+// |	└───────────────────┘
+func TestDeleteHistoryBranch_DeletedAncestor(t *testing.T) {
+	store, dbMock, _ := setUpMocks(t)
+
+	request := &persistence.InternalDeleteHistoryBranchRequest{
+		BranchInfo: types.HistoryBranch{
+			TreeID:   "TestTreeID",
+			BranchID: "B",
+			Ancestors: []*types.HistoryBranchRange{
+				{
+					BranchID:    "A",
+					BeginNodeID: 0,
+					EndNodeID:   2,
+				},
+			},
 		},
+		ShardID: testShardID,
+	}
+
+	expectedTreeFilter := &nosqlplugin.HistoryTreeFilter{
+		ShardID:  testShardID,
+		TreeID:   "TestTreeID",
+		BranchID: common.Ptr("B"),
+	}
+
+	expectedNodeFilters := []*nosqlplugin.HistoryNodeFilter{
 		{
 			ShardID:   testShardID,
 			TreeID:    "TestTreeID",
-			BranchID:  "TestAncestorBranchID",
-			MinNodeID: 7,
+			BranchID:  "B",
+			MinNodeID: 2,
+		},
+		{
+			ShardID:   testShardID,
+			TreeID:    "TestTreeID",
+			BranchID:  "A",
+			MinNodeID: 3,
 		},
 	}
 
-	// Expect to delete the history branch
-	dbMock.EXPECT().DeleteFromHistoryTreeAndNode(gomock.Any(), expecedTreeFilter, expectedNodeFilters).
+	dbMock.EXPECT().DeleteFromHistoryTreeAndNode(gomock.Any(), expectedTreeFilter, expectedNodeFilters).
 		Return(nil).Times(1)
 
 	historyTree := []*nosqlplugin.HistoryTreeRow{
 		{
 			ShardID:  testShardID,
 			TreeID:   "TestTreeID",
-			BranchID: "TestBranchID",
+			BranchID: "B",
 			Ancestors: []*types.HistoryBranchRange{
 				{
 					BranchID:    "TestAncestorBranchID",
```

Comment:
should be `"A"`, this was confusing me for a while (talked IRL, and I'm convinced it's not a sign of deeper issues now that I understand the other values better.  just writing it down for visibility)